cmd/csi-snapshotter/main.go

```diff
@@ -60,10 +61,14 @@ var (
 	snapshotNamePrefix              = flag.String("snapshot-name-prefix", "snapshot", "Prefix to apply to the name of a created snapshot")
 	snapshotNameUUIDLength          = flag.Int("snapshot-name-uuid-length", -1, "Length in characters for the generated uuid of a created snapshot. Defaults behavior is to NOT truncate.")
 	showVersion                     = flag.Bool("version", false, "Show version.")
+
+	leaderElection          = flag.Bool("leader-election", false, "Enables leader election.")
+	leaderElectionNamespace = flag.String("leader-election-namespace", "", "The namespace where the leader election resource exists. It should be the same namespace where the attacher runs.")
```

Comment:
I would much prefer it if we tried to discover the namespace instead of having the user set it. Left this here to be consistent with external attacher

Reply:
Can namespace be passed in as an ENV variable? then you can use downward API to get the namespace.  I guess you can also set the argument via downward API already.  I'm unsure if we want a mix of ENV and args.

Reply:
That's fair. I'll leave this for now.

Reply:
external-provisioner discovers the namespace via serviceaccount token:

https://github.com/kubernetes-csi/external-provisioner/blob/fd73d0d7a28b894aacecf0c8749db53c9cd27456/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/controller/controller.go#L1388-L1401

IMO, it's not so bad to follow. Adding it to csi-lib-util could be useful.

Reply:
Why does the namespace for snapshotter have to be in the same namespace as the attacher? Snapshotter could be running without an attacher.

Reply:
Do we actually need the namespace field for snapshotter?

Only provisioner and attacher have legacy leader election that we may need backwards compat for.

Reply:
Looking at this merged PR for resizer, it says namespace "Defaults to the pod namespace if not set"?  I am confused about the namespace.
https://github.com/kubernetes-csi/external-resizer/pull/31/files#diff-6bd6f3383abbcca8c0512632cea597e5R47

Reply:
Probably "Defaults to the pod namespace if not set" is still the right comment here too.  The pod is where the leader election resource runs.

Reply:
fixed